### PR TITLE
feat: update push notification API request/response

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

Push Notification API update. We now support a platform field with the registration tokens.
This allows better push notification support between our different platforms.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24542?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

This has no impact. The notification feature is not released yet.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/push-platform-notifications/services/services.ts [hunk 1]
```diff
@@ -10,9 +10,14 @@ const url = process.env.PUSH_NOTIFICATIONS_SERVICE_URL;
 const REGISTRATION_TOKENS_ENDPOINT = `${url}/v1/link`;
 const sw = self as unknown as ServiceWorkerGlobalScope;
 
+export type RegToken = {
+  token: string;
+  platform: 'extension' | 'mobile' | 'portfolio';
+};
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/push-platform-notifications/services/services.ts:16]: New request/response shape, just needed updating of the implementation and tests.

Comment (matteoscurati) [reply app/scripts/controllers/push-platform-notifications/services/services.ts:16]: It's probably not worth using a constant, right?

**Review by matteoscurati — APPROVED** [review]
LGTM :) (just a nit comment about the new RegToken type)

File: app/scripts/controllers/push-platform-notifications/services/services.ts (hunk 1 above)
Comment (Prithpal-Sooriya) [reply app/scripts/controllers/push-platform-notifications/services/services.ts:16]: yah, we can extract this to shared libraries in future :P

**Review by danjm — APPROVED** [review]

**Pull request merged**